basics/uri.adoc

```diff
@@ -25,253 +26,488 @@ specific language governing permissions and limitations
-* Path, query, and message *MUST* be case-sensitive
+NOTE: The ABNF fragments above only contain the altered productions.
 
-If a segment in the path is empty:
+.REQ~LONG-FORM-AUTHORITY-MAPPING
+[#req~long-form-authority-mapping~1]
+--
+A long form URI's _authority_ *MUST* be mapped to/from the UUri's `authority.name` property following the rules defined in link:https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2[RFC3986, Section 3.2.2]. In particular, the URI *MUST NOT* contain an _authority_ if `authority.name` is empty and vice versa.
+--
 
-* The path *MUST NOT* start with "//"
-* Path *MUST* be https://www.rfc-editor.org/rfc/rfc3986#section-3.3[path-abempty] (i.e. its value can be either an absolute path or empty)
 
-NOTE: Local URIs do not contain the authority and begin with `/` (forward slash)
+.REQ~LONG-FORM-PATH-MAPPING
+[#req~long-form-path-mapping~1]
+--
+A long form URI's _path_ *MUST* be mapped to/from the UUri's `entity.name`, `entity.version_major`, `resource.name` and `resource.instance` properties.
 
+Each property value *MUST* be mapped to a _segment_ following the rules defined in link:https://datatracker.ietf.org/doc/html/rfc3986#section-3.3[RFC3986, Section 3.3]. An empty property *MUST* be mapped to an empty segment. 
 
-=== Short URIs
+The `entity.version_major` *MUST* be mapped to the decimal string representation of the property value, e.g. a (hex) value of `0x1a` is mapped to the (decimal) string literal `26`.
 
-Short URIs are UUri objects that have been serialized to a string containing only numeric values of a UUri.
+The `resource.name` and `resource.instance` *MUST* be mapped to a single _segment_ using the following production:
 
-.Short Form URI Description
-[#short-form-uri]
-[width="00%",cols="23%,18%,59%",options="header"]
-|===
-|Item |Value |Description
+[source,abnf]
+----
+resource = [ resource.name [ '.' resource.instance ]]
+----
 
-|*`SCHEME`* |`up:` |Scheme (protocol name)
-|*`UAUTHORITY_IPV4`* |unreserved |IPv4 address per https://datatracker.ietf.org/doc/html/rfc791[RFC791].
+NOTE The `resource` _segment_ will be empty if `resource.name` is empty.
+--
 
-|*`UAUTHORITY_IPV6`* |unreserved |IPv6 address per https://datatracker.ietf.org/doc/html/rfc4291[RFC4291].
+.REQ~LONG-FORM-FRAGMENT
+[#req~long-form-fragment~1]
+--
+A long form URI's _fragment_ *MUST* be empty, if any of the UUri's `resource.name` or `resource.message` are empty.
 
-|*`UAUTHORITY_ID`* |unreserved a| Identifier
+Otherwise, a long form URI's _fragment_ *MUST* be mapped to/from the UUri's `resource.message` property following the rules defined in https://datatracker.ietf.org/doc/html/rfc3986#section-3.5[RFC3986, Section 3.5].
+--
 
- *MAY* be a valid vehicle VIN (17 MSB)
- *MUST* fit within 255 bytes (when used for micro form)
+.REQ~LONG_FORM_VALIDATION
+[#req~long-form-validation~1]
+--
+[sidebar]
+Is this needed? It currently only verifies that the invariants hold (which, by definition, they should do anyway) ...
```

Comment:
If you build a UUri from Short form string then this would be false as there is no name for entity or resource

Reply:
Is this about the following?
1. Short form URI `up://my-vin/1234/1/5678` gets deserialized into a UUri
2. UUri shall now be serialized to a long form URI but that would result in an empty URI because the entity.id and resource.id may not have been resolved to their corresponding names

I see your point but it (again) raises the question if we allow empty properties during serialization or not. The latter would mean that we do not support generating URIs that contain wildcards and that serialization behavior is inconsistent with deserialization, where we (want to?) support wilcards (i.e. empty path segments) ...